libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -441,6 +438,28 @@ impl Greenlight {
         Ok(on_chain_balance)
     }
 
+    async fn pending_onchain_balance(
+        &self,
+        peer_channels: &[cln::ListpeersPeersChannels],
+    ) -> Result<u64> {
+        let pending_onchain_balance = peer_channels.iter().fold(0, |a, b| match b.state() {
+            ChanneldShuttingDown | ClosingdSigexchange | ClosingdComplete | AwaitingUnilateral
+            | FundingSpendSeen => a + b.min_to_us_msat.clone().unwrap_or_default().msat,
+            Onchain => {
+                if b.closer() == cln::ChannelSide::Local
+                    && b.status[b.status.len() - 1].contains("DELAYED_OUTPUT_TO_US")
+                {
+                    a + b.min_to_us_msat.clone().unwrap_or_default().msat
+                } else {
+                    a
+                }
+            }
```

Comment:
Can you explain why 'onchain' gets a specific case?
1) 'Onchain' means this was a unilateral close attempt by the peer. In that case, you should not get a delayed output? If you initiate the unilateral close yourself (AwaitingUnilateral), you get a delayed output. Or does this have to do when htlcs are involved?

2) Why is the balance of the delayed outputs added to the pending balance, but not the balance of the non-delayed outputs?

Reply:
1. When the user unilateral closes the channel it will as you write get a status as `AwaitingUnilateral` first, but when the closing transaction is confirmed onchain the channel gets a status as `Onchain`. Though for the peer causing they'll have to wait X amount of blocks before they can spend those funds, which is reflected in the status.

2. Adding the balance of the delayed outputs to the pending balance since they cannot be spent. Also if earlier states shows a pending balance it makes sense to add them here too as the funds cannot be spent due to the delayed outputs. Adding the balance of the non delayed outputs does not make sense to add as a pending balance since they can be spent.

Reply:
OK thanks, makes sense.
Maybe add a comment in the code there to explain what it's doing, because it's hard to figure out frmo cln documentation why we're doing what we're doing here.

Reply:
Good point, added a comment